Src/F_Interfaces/AmrCore/AMReX_fillpatch_mod.F90

```diff
@@ -184,6 +201,135 @@ subroutine amrex_fillpatch_two (mf, told_c, mfold_c, tnew_c, mfnew_c, geom_c, fi
          &                      pre_interp_ptr, post_interp_ptr)
   end subroutine amrex_fillpatch_two
 
+  subroutine amrex_fillpatch_two_faces(mf, told_c, mfold_c, tnew_c, mfnew_c, geom_c, fill_physbc_cx, &
+#if AMREX_SPACEDIM>1
+        &                                   fill_physbc_cy, &
+#if AMREX_SPACEDIM>2
+        &                                   fill_physbc_cz, &
+#endif
+#endif
+        &                                   told_f, mfold_f, tnew_f, mfnew_f, geom_f, fill_physbc_fx, &
+#if AMREX_SPACEDIM>1
+        &                                   fill_physbc_fy, &
+#if AMREX_SPACEDIM>2
+        &                                   fill_physbc_fz, &
+#endif
+#endif
+        &                               time, scomp, dcomp, ncomp, rr, interp, lo_bc, hi_bc, &
+        &                               pre_interp, post_interp)
+    type(amrex_multifab), intent(inout) :: mf(amrex_spacedim)
+    type(amrex_multifab), intent(in   ) :: mfold_c(amrex_spacedim), mfnew_c(amrex_spacedim)
+    type(amrex_multifab), intent(in   ) :: mfold_f(amrex_spacedim), mfnew_f(amrex_spacedim)
+    integer, intent(in) :: scomp, dcomp, ncomp, rr, interp
+    integer, dimension(amrex_spacedim,scomp+ncomp-1), target, intent(in) :: lo_bc, hi_bc
```

Comment:
We have an issue here.  This should be a 3D array, `(amrex_spacedim,amrex_spacedim,scomp+ncomp-1)`.  Let's suppose `scomp = 1` and `ncomp = 1`, then we need 9, not 3 ints to specify the lower BC, because there are lo-x, lo-y and lo-z directions for x-, y- and z-faces.  I think we should simplify this and assume that there is only one component and scomp = 1 so that we can drop the component dimension.

Reply:
I agree with the intent changes in the PR. I'm double-checking that the new FillPatch uses 9 correctly as is. Then, we can either do this simplification or take the time to make it robust? Depends on how difficult it is, I suppose.